kubetest/main.go

```diff
@@ -396,9 +396,19 @@ func installGcloud(tarball string, location string) error {
 
 func prepareGcp(kubernetesProvider string) error {
 	// Ensure project is set
+	var err error
 	p := os.Getenv("PROJECT")
 	if p == "" {
-		return fmt.Errorf("KUBERNETES_PROVIDER=%s requires setting PROJECT", kubernetesProvider)
+		log.Printf("K8s provider: %v, PROJECT not set, will make a lease from boskos.", kubernetesProvider)
+		p, err = boskos(timeout)
```

Comment:
Hmm.. I think we can do better than this. Let's iterate on the design of this interface in the design doc before implementing it

I would like to see something more like this:

```
URL := "http://boskos"
boskos := Boskos(URL)
lease, err := boskos.Lease("project", "free", "leased", 5 * time.Minute)
err := boskos.Extend(lease, 5 * time.Minute)
if testsFailed {
  err := boskos.Return(lease, "hospitalized")
} else {
  err := boskos.Return(lease, "dirty")
}
```

Specifically I expect our janitor to use the same pattern (and same library!) as kubetest. The only difference is that kubetest leases clean projects and returns them as dirty. Whereas the janitor looks for dirty tests and returns them as clean.

We also need a way to create a hospital and a process to ensure we do not put too many things into them. Most likely by some mechanism to sets a ceiling on the number and/or age of hospitalized jobs.

But regardless I think this is interesting and complex enough to make it worthwhile to spend a few hours writing down this design and discussing it before implementing.